data_filter_mongodb/internal/mongo/mongo_mock.go

```diff
@@ -0,0 +1,71 @@
+package mongo
+
+import (
+	"fmt"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.uber.org/zap"
+	"log"
+	"net/http"
+)
+
+type MockMongoClientStruct struct {
+	Mongo *mongo.Client
+	Database string
+	ClientOptions *options.ClientOptions
+	Logger *zap.Logger
+}
+
+func NewMockMongoClient(mongo *mongo.Client, database string) MockMongoClient {
+	logger, err := zap.NewProduction()
+	if err != nil {
+		log.Fatalf("Failed to create zap logger %v", err)
+	}
+	return &MockMongoClientStruct{
+		Mongo:    mongo,
+		Database: database,
+		Logger: logger,
+	}
+}
+
+func (c *MockMongoClientStruct) GetAllDataFromMongo(w http.ResponseWriter) {
+	panic("implement me")
+}
+
+type MockMongoClient interface {
+	QueryMongo(pipeline []Queries) Result
+	GetAllDataFromMongo(w http.ResponseWriter)
+}
+
+const (
+	// data for successful case
+	johnDanerys = "[{map[$and:[map[name:map[$eq:danerys]] map[name:map[$eq:john]]]]} {map[$and:[map[manager:map[$eq:danerys]] map[name:map[$eq:john]]]]}]"
+	// data for unsuccessful case
+	johnJamie =  "[{map[$and:[map[name:map[$eq:jamie]] map[name:map[$eq:john]]]]} {map[$and:[map[manager:map[$eq:jamie]] map[name:map[$eq:john]]]]}]"
+	)
+
+func (c *MockMongoClientStruct) QueryMongo(pipeline []Queries) Result {
+	var r Result
+	fmt.Println(pipeline)
+	if johnDanerys == fmt.Sprintf("%s",pipeline) {
+		r = Result{
+			Defined: true,
+			Data: []EmployeeDetails{
+				{
+					Name:        "john",
+					Designation: "Software Engineer",
+					Salary:      70000,
+					Email:       "john@opa.com",
+					Mobile:      "7436238746",
+					Manager:     "danerys",
+				},
+			},
+		}
+	} else if johnJamie == fmt.Sprintf("%s", pipeline) {
+		r = Result{
+			Defined: true,
+			Data:    nil,
+		}
+	}
+	return r
+}
```

Comment:
I don't understand why we are hardcoding the Mongo queries.

Reply:
We are hardcoding the dataset that we expect from MongoDB. we aren't hardcoding queries. It's expected data from MongoDB.